pkg/controller/perconaservermongodb/backup.go

```diff
@@ -364,11 +367,115 @@ func secretExists(ctx context.Context, cl client.Client, nn types.NamespacedName
 	var secret corev1.Secret
 	err := cl.Get(ctx, nn, &secret)
 	if err != nil {
-		if k8serrors.IsNotFound(err) {
+		if k8sErrors.IsNotFound(err) {
 			return false, nil
 		}
 		return false, err
 	}
 
 	return true, nil
 }
+
+func isPodUpToDate(pod *corev1.Pod, stsRevision, image string) bool {
+	if pod.Labels["controller-revision-hash"] != stsRevision {
+		return false
+	}
+
+	for _, ct := range pod.Spec.Containers {
+		if ct.Name == naming.ContainerBackupAgent {
+			if ct.Image != image {
+				return false
+			}
+		}
+	}
+
+	return true
+}
+
+func (r *ReconcilePerconaServerMongoDB) reconcileBackupVersion(ctx context.Context, cr *api.PerconaServerMongoDB) error {
+	log := logf.FromContext(ctx)
+
+	if !cr.Spec.Backup.Enabled {
+		return nil
+	}
+
+	if cr.Status.State != api.AppStateReady {
+		return nil
+	}
+
+	if cr.Status.BackupVersion != "" && cr.Status.BackupImage == cr.Spec.Backup.Image {
+		return nil
+	}
+
+	var rs *api.ReplsetSpec
+	for _, r := range cr.Spec.Replsets {
+		rs = r
+		break
+	}
```

Comment:
If for some reason `cr.Spec.Replsets` is zero, rs will be nil and this will cascade to the remaining functionality. Maybe it is a more suitable approach to go with something like `len(cr.Spec.Replsets) >0` and then assign the first replica, otherwise return an error.